mysys/my_open.c

```diff
@@ -96,15 +102,13 @@ int my_close(File fd, myf MyFlags)
     my_errno=errno;
     if (MyFlags & (MY_FAE | MY_WME))
       my_error(EE_BADCLOSE, MYF(ME_BELL | ME_WAITTANG | (MyFlags & (ME_JUST_INFO | ME_NOREFRESH))),
-               my_filename(fd),errno);
+               name,errno);
   }
-  if ((uint) fd < my_file_limit && my_file_info[fd].type != UNOPEN)
+  if (name)
   {
-    my_free(my_file_info[fd].name);
-    my_file_info[fd].type = UNOPEN;
+    my_free(name);
   }
```

Comment:
What's the point of freeing it after `close()`?

Reply:
Did you actually verify that `my_close()` is never called concurrently?

Reply:
The existing implementation would have also generated a EBADF (Errno 9) error logged if this occurred. EBADF hasn't existed according to JIRA search and only MDEV-13646 shows Errno 9 (and it doesn't look like a `close` call). Nothing recent in bugs.mysql.com either.